### PR TITLE
Add transparent refresh of Auth Token

### PR DESCRIPTION
**Pull request by caleblloyd**
- Injects JS onto HTML pages to transparently refresh AUTH_TOKEN if it is within 5 minutes of expiring
- Add environment variables for `UPSTREAM_SERVER`, `COOKIE_NAME`, and `INJECT_REFRESH_JS`
- Pass `X-Authorize-Sub` header to upstream server
- Update readme

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -205,16 +278,116 @@ func callbackHandler(w http.ResponseWriter, r *http.Request, conf config) {
...
+}
+
+func errorHandler(w http.ResponseWriter, r *http.Request, conf *config) {
+	params := r.URL.Query()
+	ssoErr := params.Get("error")
+
+	w.WriteHeader(http.StatusUnauthorized)
+
+	io.WriteString(w, `
+	<!DOCTYPE html>
```
Comment (colek42) [inline server.go]: Can we put this in a "const"?

File: stage/var/okta-nginx/refresh.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/// refresh.js
+/// this code is minifed by refresh-minify.sh
```
Comment (colek42) [inline stage/var/okta-nginx/refresh.js]: lol!  This is awesome.

File: server.go [hunk 3]
```diff
@@ -35,7 +37,7 @@ type jwtResponse struct {
 	AccessToken string `json:"access_token"`
 }
 
-func getConfig() config {
+func getConfig() *config {
```
Comment (colek42) [inline server.go]: good catch

File: server.go (hunk 1 above)
Comment (caleblloyd) [reply server.go]: I don't think so, the `conf.appOrigin` variable is part of the string so it can't be const

File: server.go [hunk 4]
```diff
@@ -205,16 +278,116 @@ func callbackHandler(w http.ResponseWriter, r *http.Request, conf config) {
...
+	expFloat, ok := exp.(float64)
+	if !ok {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("refreshHandler: Unable to convert 'exp' to float64")
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	if expFloat-float64(time.Now().UTC().Unix()) < (5 * time.Minute).Seconds() {
+		io.WriteString(w, redirectURL)
```
Comment (colek42) [inline server.go]: err

**Pull request merged**